fetch/api/cors/data-url-shared-worker.html

```diff
@@ -0,0 +1,59 @@
+<!DOCTYPE html>
+<html>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+
+const fetch_from_data_url_worker_test =
+    (url, cors, expectation, description) => {
+  promise_test(async () => {
+    const fetchURL = new URL(url, location.href) +
+        `${cors === 'null-origin'
+             ? '?pipe=header(Access-Control-Allow-Origin, null)' : ''}`;
+    const scriptURL =
+      `data:text/javascript,` +
+      `async function test(port) {` +
+      `  let allowed = true;` +
+      `  try {` +
+      `    await fetch('${fetchURL}');` +
+      `  } catch (e) {` +
+      `    allowed = false;` +
+      `  }` +
+      `  port.postMessage({allowed});` +
+      `}` +
+      `onconnect = e => {` +
+      `  test(e.ports[0]);` +
+      `};`;
+    const worker = new SharedWorker(scriptURL);
+    const msgEvent =
+        await new Promise(resolve => worker.port.onmessage = resolve);
+    assert_equals(msgEvent.data.allowed ? 'allowed' : 'rejected', expectation);
+  }, description);
+};
+
+fetch_from_data_url_worker_test(
+  '../resources/top.txt',
+  'no-cors',
+  'rejected',
+  'fetching "top.txt" with no CORS should be rejected.'
+);
+fetch_from_data_url_worker_test(
+  '../resources/top.txt',
+  'null-origin',
+  'allowed',
+  'fetching "top.txt" with CORS allowing null origin should be rejected.'
+);
+fetch_from_data_url_worker_test(
+  'data:text/plain, top',
+  'no-cors',
+  'allowed',
+  'fetching data url script with no CORS should be allowed.'
+);
+fetch_from_data_url_worker_test(
+  'data:text/plain, top',
+  'null-origin',
```

Comment:
This is no-op, because `data:` URL doesn't handle `pipe=header(...`.

Reply:
Thanks. I removed the test to fetch data url scripts with cors.